src/Devito.jl

```diff
@@ -2274,8 +1840,12 @@ export Function, SparseFunction, SparseTimeFunction, SubDomain, TimeFunction, ap
 export backward, ccode, configuration, configuration!, switchconfig, coordinates, coordinates_data
 export data, data_allocated, data_with_halo, data_with_inhalo, dimension, dimensions
 export dx, dy, dz, evaluate, extent, forward, grid, halo, indexed, inject, interpolate
-export localindices, localindices_with_halo, localindices_with_inhalo, localsize, name
+export localindices, localindices_with_halo, localindices_with_inhalo, name
 export nsimplify, origin, size_with_halo, simplify, solve, space_order, spacing, spacing_map
 export step, subdomains, subs, thickness, value, value!
 
+if !isdefined(Base, :get_extension)
```

Comment:
I get the intention, but it won't work, that's why it needs that annoying Requires workaround. `MPI` is not a dependency, so this will error with a missing dependency if it tries to include it.

Reply:
I was trying to follow this: https://pkgdocs.julialang.org/v1/creating-packages/#Transition-from-normal-dependency-to-extension.  I guess I need to add MPI to the [deps] section.  Anyways, since Julia 1.10 is lts, not sure how much to worry about it.  We only test for 1.10 and latest.

Reply:
actually, I just noticed that Devito.jl seems to require Julia >= 1.9 anyways.  So, I guess it is safe to remove that conditional entirely.